File: python/general/164. 最大间距.py
from typing import List
class Solution:
    def maximumGap(self, nums: List[int]) -> int:
        if len(nums) == 1 or not nums:
            return 0
        nums.sort()
        res = -1
        for i in range(len(nums)-1):
            j = i+1
            res = max(abs(nums[j] -nums[i]),res)
        return res
s = Solution()
nums = [3,6,9,1]
print(s.maximumGap(nums))

File: go/general/164. 最大间距.go
package main

import "sort"

func maximumGap(nums []int) int {
	if len(nums) == 0 || len(nums) == 1 {
		return 0
	}
	sort.Ints(nums)
	res := -1
	for i := 0; i < len(nums)-1; i++ {
		j := i + 1
		if nums[j]-nums[i] >= 0 {
			if res < nums[j]-nums[i] {
				res = nums[j] - nums[i]
			}
		} else {
			if res < nums[i]-nums[j] {
				res = nums[i] - nums[j]
			}
		}
	}
	return res
}
